dask/dataframe/utils.py

```diff
@@ -635,7 +635,7 @@ def assert_sane_keynames(ddf):
         while isinstance(k, tuple):
             k = k[0]
         assert isinstance(k, (str, bytes))
-        assert len(k) < 100
+        assert len(k) < 150
```

Comment:
What's causing key names to get large? While 100 is fairly arbitrary, we don't want key names to get too unwieldy

Reply:
Just a bunch of sub-tasks getting concatenated together. If memory serves it was just 101 in this case, but the key name still looked reasonable to me, so I bumped the check.

Reply:
The keys triggering this assert look like 

```
dataframe-groupby-count-chunk-81b2830612fbebe66693c0556d98d6a7-index-cef97a32d3f76569b6c7ff29b6d2a3df
```

which seem to contain multiple hashes (we probably can make due with one). I've not yet pinpointed where these are originating from

Reply:
Thoughts on this? 

```diff
diff --git a/dask/dataframe/core.py b/dask/dataframe/core.py
index cc82fd791..49ba5c153 100644
--- a/dask/dataframe/core.py
+++ b/dask/dataframe/core.py
@@ -562,7 +562,7 @@ Dask Name: {name}, {layers}"""
         return self.map_partitions(
             getattr,
             "index",
-            token=self._name + "-index",
+            token=key_split(self._name) + "-index",
             meta=self._meta.index,
             enforce_metadata=False,
         )
diff --git a/dask/dataframe/utils.py b/dask/dataframe/utils.py
index f98e98f3e..f66212de4 100644
--- a/dask/dataframe/utils.py
+++ b/dask/dataframe/utils.py
@@ -635,7 +635,7 @@ def assert_sane_keynames(ddf):
         while isinstance(k, tuple):
             k = k[0]
         assert isinstance(k, (str, bytes))
-        assert len(k) < 150
+        assert len(k) < 100
         assert " " not in k
         assert k.split("-")[0].isidentifier(), k
```

Reply:
At first I thought adding `key_split` would cause us to loose information in the final hash we're using for this key, but I convinced myself this actually isn't the case since we pass `self` (the dask dataframe instance) to the internal `tokenize(...)` used in `map_partitions`. This should make sure we're including the original hash in `self._name` when creating the final hash that's used for the output dask key.

Reply:
Seems safe to me, seeing as the underlying `_Frame` gets tokenized in `map_partitions`, and this `token` only serves as a prefix.

Reply:
> At first I thought adding `key_split` would cause us to loose information in the final hash we're using for this key, but I convinced myself this actually isn't the case since we pass `self` (the dask dataframe instance) to the internal `tokenize(...)` used in `map_partitions`. This should make sure we're including the original hash in `self._name` when creating the final hash that's used for the output dask key.

Ah, missed this comment. Looks like we came to the same conclusion